### PR TITLE
Integrate Quick Add Work Item in Board View

### PR DESCRIPTION
**Pull request by SMahil**
Integrate Quick Add Work Item in Board View #582

### CODE REVIEW

**Comment by ldimaggi** [comment]
I just tried to build and run the UI (I wanted to create the page object model for the board display) from this PR - the "work" tab display is empty and this error is logged:

ERROR in ./src/app/work-item/work-item-quick-add/work-item-quick-add.component.ts
(51,22): error TS2339: Property 'exmp' does not exist on type 'WorkItemQuickAddComponent'.

Am I missing something? Thx!

**Comment by SMahil** [comment]
@ldimaggi there was one statement to check the output in console . I forgot to remove it earlier but now it is removed

File: src/app/work-item/work-item-board/work-item-board.component.html [hunk 1]
```diff
@@ -8,43 +8,48 @@
...
-              </ul>
-            </aside>
-            <aside>
-              <div class="col-md-10 col-sm-10 col-xs-9">
-                <p class="multiline-truncation">{{ item.attributes['system.title'] }}</p>
-              </div>
-              <div class="col-md-2 col-sm-2 col-xs-3 no-padding user-avatar">
-                <img *ngIf="item.relationalData?.assignees?.length"
+
+          <div class="boardlaneWrapper">
```
Comment (vikram-raj) [inline src/app/work-item/work-item-board/work-item-board.component.html]: Don't use camel case format for the class name. Always use a delimiter-separated word and delimiter should be a hyphen.

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div class="quickAdd_container" id="workItemQuickAdd_container" >
+<div class="quickAdd_container" id="workItemQuickAdd_container" *ngIf="wilistview === 'wi-list-view'" >
```
Comment (vikram-raj) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: change ID and class name to delimiter-separated word in this file.

File: src/app/work-item/work-item-board/work-item-board.module.ts [hunk 3]
```diff
@@ -3,17 +3,18 @@ import { CommonModule }       from '@angular/common';
...
 //pipes
 
 import { AlmFilterBoardList } from './../../pipes/alm-board-filter.pipe';
 
 
 
 @NgModule({
-  imports:      [ AlmIconModule, BoardRoutingModule, CommonModule],
-  declarations: [ AlmFilterBoardList, CountPipe , WorkItemBoardComponent, ],
+  imports:      [ AlmIconModule, BoardRoutingModule, CommonModule , WorkItemQuickAddModule],
```
Comment (vikram-raj) [inline src/app/work-item/work-item-board/work-item-board.module.ts]: break it into new lines.

File: src/app/work-item/work-item-board/work-item-board.component.html (hunk 1 above)
Comment (SMahil) [inline src/app/work-item/work-item-board/work-item-board.component.html]: @vikram-raj We have followed the camel-case convention for naming classnames and for ID we  mentioned the component name for better understanding

File: src/app/work-item/work-item-board/work-item-board.module.ts (hunk 3 above)
Comment (SMahil) [inline src/app/work-item/work-item-board/work-item-board.module.ts]: Is there any specific reason for breaking it into two lines?

File: src/app/work-item/work-item-board/work-item-board.component.html (hunk 1 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-board/work-item-board.component.html]: @SMahil Yeah, I think Vikram is right. You can anyway use board-lane-wrapper instead. Css classes should not be named in camel cases.

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html (hunk 2 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: Let them be like this for now in this line, because they are not part of this PR.

Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: But we should definitely avoid using camel case on IDs and Classes.

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html [hunk 4]
```diff
@@ -32,4 +32,40 @@
       <span (click)="toggleQuickAdd()" class="pull-right closeQuickAdd"><i class="pficon pficon-close"></i></span>
     </div>
   </div>
+</div>
+
+<!-- Quick ADD Template for Board View -->
+<div class="quickAdd_container quickAdd_container2" id="workItemQuickAdd_container" *ngIf="wilistview === 'wi-card-view'" >
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: Why do we have two of these classes? Can we combine them into one?

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html (hunk 2 above)
Comment (SMahil) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: We have camel case pattern at all places. If we dont want camel case in IDs and Classes it should be a separate issue

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html (hunk 4 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: Also change the name to quick-add-container

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 5]
```diff
@@ -87,6 +74,70 @@
         }
       }
     }
+    .icon-btn, .icon-btn-disabled{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Please add a space before { where ever it's not there.

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 6]
```diff
@@ -87,6 +74,70 @@
...
+      cursor: pointer;
+    }
+    .icon-btn-disabled{
+      color: $color-pf-black-300;
+      cursor: default;
+    }
+  }
+}
+
+.quickAdd_container2{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Combine this class with **quickAdd_Container** and change the name to **quick-add-container**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 7]
```diff
@@ -87,6 +74,70 @@
...
+  position: relative;
+  width: $width100;
+  float: left;
+  @include Border(0 , solid , transparent);
+  .addWorkItem{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **add-work-item**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 8]
```diff
@@ -87,6 +74,70 @@
...
+    background: $color-pf-white;
+    margin: em(5) 0;
+    @include Border(1px, solid ,$color-pf-black-400);
+    text-align: $textCenter;
+    padding: em(10) em(15);
+    .workItemQuickAdd_saveBtn{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **work-item-quick-add-save-btn**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 9]
```diff
@@ -87,6 +74,70 @@
...
+      i{
+        font-size: em(18);
+        color: $color-pf-blue-400;
+      }
+    }
+  }
+  .addWorkItemBlk{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **add-work-item-blk**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 10]
```diff
@@ -87,6 +74,70 @@
...
+    margin-top: em(5);
+    @include Border(1px, solid ,$color-pf-black-300);
+    .quickAdd_temp2-Story{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **quick-add-temp2-story**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 11]
```diff
@@ -87,6 +74,70 @@
...
+      float: left;
+      padding: em(2);
+      @include Border(1px, solid ,$color-pf-black-700);
+      i{
+        margin: 0;
+        font-size: em(12);
+      }
+    }
+    .quickAdd_temp2-Title{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **quick-add-temp2-title**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 12]
```diff
@@ -87,6 +74,70 @@
...
+    }
+  }
+  .addWorkItemBlk{
+    margin-top: em(5);
+    @include Border(1px, solid ,$color-pf-black-300);
+    .quickAdd_temp2-Story{
+      float: left;
+      padding: em(2);
+      @include Border(1px, solid ,$color-pf-black-700);
+      i{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Is it a good way to do it? What you use a class name instead?

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 13]
```diff
@@ -87,6 +74,70 @@
...
+        font-size: em(12);
+      }
+    }
+    .quickAdd_temp2-Title{
+      padding: 0 em(25);
+      input{
+        margin-bottom: em(7);
+      }
+    }
+    .workItemQuickAdd_Add{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **work-item-quick-add**

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 14]
```diff
@@ -87,6 +74,70 @@
...
+      font-size: em(12);
+      @include Border(1px, solid ,$color-pf-black-300);
+      margin-top: em(10);
+      padding: em(1);
+    }
+    .closeQuickAdd{
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: Change the class name to **close-quick-add**

File: src/app/work-item/work-item-board/work-item-board.module.ts (hunk 3 above)
Comment (vikram-raj) [inline src/app/work-item/work-item-board/work-item-board.module.ts]: Just for readability and better coding style. A single should not cross the max limit of 80 characters in a line.

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html (hunk 4 above)
Comment (SMahil) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: there are only few css elements which have different styles from first template. thats why keeping two classes

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss (hunk 6 above)
Comment (SMahil) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: two classes are needed over here

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss (hunk 12 above)
Comment (SMahil) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: we can use where we are sure there will no repetition for the tag

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.html (hunk 4 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.html]: Make separate classes for those two in this case. Only **quick-add_container2** for this one, do not use **quickAdd_container**.

**Comment by sanbornsen** [comment]
screen shot 2017-01-09 at 1 34 13 pm
The distance between the cards are not the same. Is this how the UX is designed?

**Comment by sanbornsen** [comment]
Adding an item is not reflecting on the list.

**Comment by SMahil** [comment]
@sanbornsen its reflecting when we are refreshing the page. same issue like issue no 637

**Comment by debloper** [comment]
The add new WI box has a 5px margin on top, but none at the bottom. All other WI cards have 5px margin on top AND bottom. Hence the issue. @sanbornsen @SMahil

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss [hunk 15]
```diff
@@ -87,6 +74,70 @@
...
+    padding: em(10) em(15);
+    .workItemQuickAdd_saveBtn{
+      i{
+        font-size: em(18);
+        color: $color-pf-blue-400;
+      }
+    }
+  }
+  .addWorkItemBlk{
+    margin-top: em(5);
```
Comment (debloper) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.scss]: This is what causing the spacing problem. `margin: em(5) auto;` should fix it.

**Comment by sanbornsen** [comment]
@SMahil yeah, I got it, but need to be fixed anyway.

**Comment by joshuawilson** [comment]
@SMahil @sanbornsen early on I had asked that all classes in a component start with the component name. I understand that convention is to use dashes not camelCase. Most of the classes use camelCase for the component name. I do not think this PR is the place to fix them all. In fact I think to be consistent we could even stay with the style for this PR. However, I do think we should consider creating a new issue for this, then updating all the classes to use the same style. Consistency is more important then convention. It improves readability.

**Comment by SMahil** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]
